submissions/pozzitive11/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,125 @@
+const cards = [
+  {
+    front: "images/front.jpg",
+    back: "images/sleepy-cat.jpg",
+    name: "sleepy",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/cat.jpg",
+    name: "cat",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/frightened-cat.jpg",
+    name: "frightened",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/brutal-cat.jpg",
+    name: "brutal",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/rock-cat.jpg",
+    name: "rock",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/cute-cat.jpg",
+    name: "cute",
+  },
+];
+
+const gameCards = cards.map(
+  ({ front, back, name }) =>
+    `<div class="game__card" id="${name}">
+             <img class="front__img" src="${front}" alt="">
+             <img class="back__img" src="${back}" alt="">
+           </div>
+           <div class="game__card" id="${name}">
+             <img class="front__img" src="${front}" alt="">
+             <img class="back__img" src="${back}" alt="">
+           </div>`
+);
+
+const gameContainer = document.querySelector(".game__container");
+gameContainer.innerHTML = gameCards.join("");
+const allGameCards = document.querySelectorAll(".game__card");
+
+let numberOfMatches = 0;
+let flippedCards = [];
+
+gameContainer.addEventListener("click", ({ target }) => {
+  const parentTarget = target.closest(".game__card");
+  if (parentTarget !== null) {
+    parentTarget.classList.add("flip");
+
+    flippedCards.push(parentTarget);
+    checkArrayOfFlippedCardsLength();
+    checkForWin();
+  }
+});
+
+function checkArrayOfFlippedCardsLength() {
+  if (flippedCards.length === 2) {
+    checkMatch();
+  }
+}
+
+function checkMatch() {
+  const match = flippedCards[0].id === flippedCards[1].id;
+
+  if (match) {
+    numberOfMatches++;
+    addingDisableClass();
+    flippedCards = [];
+    setTimeout(() => {
+      addingDisableClass();
+    }, 1000);
+  } else {
+    removeFlip();
+  }
+}
+
+function removeFlip() {
+  setTimeout(() => {
+    flippedCards[0].classList.remove("flip");
+    flippedCards[1].classList.remove("flip");
+    flippedCards = [];
+
+    addingDisableClass();
+
+    checkForWin();
+  }, 1000);
+  addingDisableClass();
+}
+
+function addingDisableClass() {
+  allGameCards.forEach((item) => {
+    item.classList.toggle("disableclick");
+  });
+}
+
+function checkForWin() {
+  if (numberOfMatches === 6) {
+    setTimeout(() => {
+      allGameCards.forEach((item) => {
+        item.classList.remove("flip");
+      });
+      alert("U WIN!");
+      flippedCards = [];
+      shuffleCards();
+      numberOfMatches = 0;
+    }, 1000);
+  }
+}
+
+function shuffleCards() {
+  allGameCards.forEach((card) => {
+    let randomPos = Math.floor(Math.random() * 12);
+    card.style.order = randomPos;
```

Comment:
Three problems.

1. There is a possibility that you will have same value of randomPos
2. You changing styles of the component in the loop. And a consequences of this changes browser will be forced to rerender page 12 times. What for?
3. You accessing styles of the dom element via `style`. Which is bad. Bad because it's quite direct and shows, that you need some hardcoded, runtime calculated values. Which is not your case.

You can shuffle array itself itself and then create cards from it.

Reply:
done